staging/src/k8s.io/dynamic-resource-allocation/structured/allocator.go

```diff
@@ -945,6 +967,20 @@ func (alloc *allocator) allocateDevice(r deviceIndices, device deviceWithID, mus
 		return false, nil, nil
 	}
 
+	// the API validation logic has in place to check the ConsumesCounter referred should exist inside SharedCounters
+	if alloc.features.PartitionableDevices && len(device.slice.Spec.SharedCounters) > 0 && len(device.basic.ConsumesCounter) > 0 {
```

Comment:
I think you can drop the `len(device.slice.Spec.SharedCounters) > 0` check. Then someone reading this code doesn't need to wonder why it's there.

In practice there never should be a situation where it's empty and `ConsumesCounter` isn't. If this ever occurs, then checking with `checkAvailableCapacity` needs to throw an error, which arguably safer than skipping the entire code.

Reply:
Removed `len(device.slice.Spec.SharedCounters) > 0` check.

Reply:
Do you still need to push your update?

Reply:
Ops. I pushed anther commit for this. It might be somehow missed while separating the changes into different commits -_-

Reply:
Now it's okay.